packages/puppeteer-core/src/common/ElementHandle.ts

```diff
@@ -249,6 +229,7 @@ export class CDPElementHandle<
     }
   }
 
+  @ElementHandle.bindIsolatedHandle
   override async screenshot(
```

Comment:
so screenshot invokes inner methods that are also decorated which results in several handles being created for each screenshot call. I think this kind of expensive given that each call adds a round-trip. Can we avoid that?

Reply:
I think the same might apply to other methods. Also, while it does not look like the screenshot needs its own isolated handle the creation of many handles seems wasteful.

Reply:
I guess this is already partially the case in the existing code though.

Reply:
we could probably count references to a handle and avoid adopting a real handle if we already have adopted one in the main world, that way we can avoid round trips.